nulib/meadow
# Use Chonky file browser with S3 data provided via GraphQL in the S3ObjectPicker component

# Summary 
Replace client-side S3 object picker with Chonky file browser

# Specific Changes in this PR
- Change GraphQL ListIngestBucketObjects to provide a folder-based listing using S3 `ListObjectsV2` API
- Add `S3ObjectProvider` React component to interact with the GraphQL backend
- Refactor `S3ObjectPicker` React component to use [Chonky](https://chonky.io/) file browser with data from `S3ObjectProvider`

# Version bump required by the PR

See [Semantic Versioning 2.0.0](https://semver.org/) for help discerning which is required.


# Steps to Test

- Check both the **Add a fileset** and **Replace fileset** features of the Preservation tab
- Make sure the available files match the work type and fileset role
- Make sure the selected file is submitted and processes through the pipeline correctly

# :rocket: Deployment Notes

**Note** - __if you check any of these boxes go to the [(always open) `main` <- `staging` PR](https://github.com/nulib/meadow/pulls) and add detailed notes and instructions to help out others who may end up deploying your changes to production__

- Backward compatible API changes
- Backwards-incompatible API changes


# Tested/Verified
